src/Domain.LinnApps/PurchaseOrders/PurchaseOrderReqService.cs

```diff
@@ -14,33 +14,38 @@ public class PurchaseOrderReqService : IPurchaseOrderReqService
 
         private readonly IPurchaseOrderReqsPack purchaseOrderReqsPack;
 
+        private readonly IRepository<PurchaseOrderReqStateChange, PurchaseOrderReqStateChangeKey> reqsStateChangeRepository;
+
         public PurchaseOrderReqService(
             IAuthorisationService authService,
-            IPurchaseOrderReqsPack purchaseOrderReqsPack)
+            IPurchaseOrderReqsPack purchaseOrderReqsPack,
+            IRepository<PurchaseOrderReqStateChange, PurchaseOrderReqStateChangeKey> reqsStateChangeRepository)
         {
             this.authService = authService;
             this.purchaseOrderReqsPack = purchaseOrderReqsPack;
+            this.reqsStateChangeRepository = reqsStateChangeRepository;
         }
 
         public void Authorise(PurchaseOrderReq entity, IEnumerable<string> privileges, int currentUserId)
         {
-            if (!this.authService.HasPermissionFor(AuthorisedAction.PurchaseOrderReqAuthorise, privileges))
+            var stage = !entity.AuthorisedById.HasValue ? "AUTH1" : "AUTH2";
+
+            if (stage == "AUTH1" && entity.State != "AUTHORISE WAIT")
```

Comment:
Req state should probably come from the db as opposed to being hardcoded strings

Reply:
Like you spotted I do have a todo for when it's changing state to get the next one from the state changes table, but for things like this where I'm checking if it's in the authorise wait state, do you think it's better to do like rough code example below?
`repo.find(x=>x.State == "AUTHORISE WAIT").State`
 It's still kinda hard coded but I guess if we changed the states it'd break the code and then we'd be forced to fix this form, which'd be better I think? There are a few bits like this hard coded in the blue req pack (Kinda need to at some point) and the old form

Reply:
Yeah, it doesn't really matter too much